lib/WasmReader/WasmParseTree.cpp

```diff
@@ -0,0 +1,132 @@
+//-------------------------------------------------------------------------------------------------------
+// Copyright (C) Microsoft Corporation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
+
+#include "WasmReaderPch.h"
+
+namespace Wasm
+{
+namespace Simd
+{
+bool IsEnabled()
+{
+#ifdef ENABLE_WASM_SIMD
+    return CONFIG_FLAG(WasmSimd);
+#else
+    return false;
+#endif
+}
+
+}
+
+namespace Threads
+{
+bool IsEnabled()
+{
+#ifdef ENABLE_WASM
+    return CONFIG_FLAG(WasmThreads) && CONFIG_FLAG(ESSharedArrayBuffer);
+#else
+    return false;
+#endif
+}
+}
+}
+
+#ifdef ENABLE_WASM
+
+namespace Wasm
+{
+namespace Simd
+{
+void EnsureSimdIsEnabled()
+{
+    if (!Wasm::Simd::IsEnabled())
+    {
+        throw WasmCompilationException(_u("Wasm.Simd support is not enabled"));
+    }
+}
+}
+
+namespace WasmTypes
+{
+
+bool IsLocalType(WasmTypes::WasmType type)
+{
+    // Check if type in range ]Void,Limit[
+#ifdef ENABLE_WASM_SIMD
+    if (type == WasmTypes::M128 && !Simd::IsEnabled())
+    {
+        return false;
+    }
+#endif
+    return (uint32)(type - 1) < (WasmTypes::Limit - 1);
```

Comment:
can you do type > Void && type < Limit? I checked and compiler generates same code.